validator/server/src/main/java/org/ethereum/beacon/validator/api/controller/DutiesController.java

```diff
@@ -0,0 +1,106 @@
+package org.ethereum.beacon.validator.api.controller;
+
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.ext.web.RoutingContext;
+import org.ethereum.beacon.chain.observer.ObservableBeaconState;
+import org.ethereum.beacon.chain.observer.ObservableStateProcessor;
+import org.ethereum.beacon.consensus.BeaconChainSpec;
+import org.ethereum.beacon.consensus.BeaconStateEx;
+import org.ethereum.beacon.core.state.ShardCommittee;
+import org.ethereum.beacon.core.types.BLSPubkey;
+import org.ethereum.beacon.core.types.EpochNumber;
+import org.ethereum.beacon.core.types.SlotNumber;
+import org.ethereum.beacon.core.types.ValidatorIndex;
+import org.ethereum.beacon.validator.api.InvalidInputException;
+import org.ethereum.beacon.validator.api.NotAcceptableInputException;
+import org.ethereum.beacon.validator.api.ValidatorDutiesService;
+import org.ethereum.beacon.validator.api.model.ValidatorDutiesResponse;
+import org.ethereum.beacon.wire.sync.SyncManager;
+import org.javatuples.Pair;
+import org.javatuples.Triplet;
+import reactor.core.publisher.Flux;
+import tech.pegasys.artemis.util.bytes.Bytes48;
+import tech.pegasys.artemis.util.uint.UInt64;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class DutiesController extends SyncRestController {
+  private final BeaconChainSpec spec;
+  private final ValidatorDutiesService service;
+  private ObservableBeaconState observableBeaconState = null;
+
+  public DutiesController(
+      ObservableStateProcessor stateProcessor,
+      ValidatorDutiesService service,
+      SyncManager syncManager,
+      BeaconChainSpec spec) {
+    super(syncManager);
+    Flux.from(stateProcessor.getObservableStateStream()).subscribe(this::updateState);
+    this.spec = spec;
+    this.service = service;
+  }
+
+  private synchronized void updateState(ObservableBeaconState observableBeaconState) {
+    this.observableBeaconState = observableBeaconState;
```

Comment:
I think in this case it would be more efficient if a local copy of `observableBeaconState` were used to read the data from, `synchronized` might be an overkill here.